gen/enum_test.go

```diff
@@ -399,3 +400,97 @@ func TestEnumAccessors(t *testing.T) {
 		})
 	})
 }
+
+func TestEnumLabelValid(t *testing.T) {
+	tests := []struct {
+		enumItem  te.EnumWithLabel
+		jsonValue string
+	}{
+		{te.EnumWithLabelUsername, `"surname"`},
+		{te.EnumWithLabelPassword, `"hashed_password"`},
+		{te.EnumWithLabelSalt, `"SALT"`},
+		{te.EnumWithLabelSugar, `"SUGAR"`},
+		{te.EnumWithLabelRelay, `"RELAY"`},
+		{te.EnumWithLabelNaive4N1, `"function"`},
+		{te.EnumWithLabel(42), `42`},
+		{te.EnumWithLabel(-1), `-1`},
+		{te.EnumWithLabel(1 << 10), `1024`},
+	}
+	for _, tt := range tests {
+		t.Run(tt.enumItem.String(), func(t *testing.T) {
+			t.Run("JSON round trip", func(t *testing.T) {
+				// marshal
+				label := te.EnumWithLabel(tt.enumItem)
+				b, err := json.Marshal(label)
+				require.NoError(t, err)
+				assert.Equal(t, []byte(tt.jsonValue), b)
+				// unmarshal
+				var gotLabel te.EnumWithLabel
+				err = json.Unmarshal(b, &gotLabel)
+				assert.Equal(t, label, gotLabel)
+			})
+			assertRoundTrip(
+				t, &tt.enumItem,
+				wire.NewValueI32(int32(tt.enumItem)),
+				"test roundtrip "+tt.jsonValue,
+			)
+		})
+	}
+}
+
+func TestEnumLabelInvalidUnmarshal(t *testing.T) {
+	tests := []struct {
+		errVal []byte
+		errMsg string
+	}{
+		{
+			[]byte("some-random-str"),
```

Comment:
this test isn't testing the `UnmarshalText` at all -- this is testing the `json.Unmarshal`. Not sure that's useful?

Reply:
https://github.com/thriftrw/thriftrw-go/blob/dev/gen/internal/tests/enums/types.go#L98
`json.Unmarshal` will call `UnmarshalText` in this test case.

I was testing against `UnmarshalText`, and was suggested to test `json.Unmarshal` in previous revision, let me know if we want to also add `UnmarshalText` here

Reply:
The issue is that this code never hits `UnmarshalText`.
https://play.golang.org/p/B1SIZYDGr_4

Using `json.Unmarshal` is fine, but the test case is incorrect since it probably should be `"some-random-str"` (e.g., the quotes should be part of the value passed to `Unmarshal`)

Reply:
ah, good catch! Thanks for this comment, I'll fix the test cases here

Reply:
actually looked at it again I added that first test case intentionally, you are right it didn't hit `UnmarshalText` at all like the rest of test cases, I added it as an invalid case for json Unmarshal, but has nothing to do with this label PR (generic invalid case), I'll make it `"some-randome-str"` so it hits `unmarshalText`